context_sync_test.go

```diff
@@ -0,0 +1,31 @@
+package honeybadger
+
+import (
+	"testing"
+	"sync"
+)
+
+func TestContextSync(t *testing.T) {
+	var wg sync.WaitGroup
+
+	instance   := newContextSync()
+	newContext := Context{"foo":"bar"}
+
+	wg.Add(2)
+
+	go update(&wg, instance, newContext)
+	go update(&wg, instance, newContext)
+
+	wg.Wait()
+
+	context := instance.internal
+
+	if context["foo"] != "bar" {
```

Comment:
I'll look into adding some extra updates/values to test the race condition. It may not be possible to prevent race conditions using this implementation, though. If you can't guarantee the order in which the goroutines are scheduled/run, then it seems that the final value will be whatever the last goroutine updated it to, right? We may need to rethink how we handle global context and possibly add a new method to add context across requests.